dicodile/update_z/tests/test_distributed_sparse_encoder.py

```diff
@@ -25,24 +26,34 @@ def test_distributed_sparse_encoder():
                   freeze_support=False, warm_start=False, random_state=27)
 
     X = rng.randn(n_channels, n_times)
-    D = rng.randn(n_atoms, n_channels, n_times_atom)
+    if not rank1:
+        D = rng.randn(n_atoms, n_channels, n_times_atom)
+    else:
+        D = rng.randn(n_atoms, n_channels+n_times_atom)
     sum_axis = tuple(range(1, D.ndim))
     D /= np.sqrt(np.sum(D * D, axis=sum_axis, keepdims=True))
-    DtD = compute_DtD(D)
+    if not rank1:
+        DtD = compute_DtD(D)
```

Comment:
you can use the same trick as the compute DtD in `alphacsc` d
for rank 1 dictionary?

Reply:
OK, so reuse/copy `compute_DtD`, `_compute_DtD_uv` and `_compute_DtD_d` from alphacsc?

Reply:
Not sure what is the best way. The dependency on `alphacsc` seems a bit unnatural but it true it would reduce the burden to only have one implementation.
Also the`alphacsc` version dies not handle 2d signals (images).

Wdyt is the best?